From 8d18088a8c4ce2a074145158a31635926a880104 Mon Sep 17 00:00:00 2001
From: Hardik Shah <hardik.shah@ti.com>
Date: Fri, 27 Feb 2009 18:00:41 +0530
Subject: [PATCH 312/364] Bug fix in alphablending and colorkeying code

Clock was not correctly get before accessing
some color keying and alphablending registers in
DSS library
---
 arch/arm/plat-omap/dss/dispc.c |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/arch/arm/plat-omap/dss/dispc.c b/arch/arm/plat-omap/dss/dispc.c
index adc2eb9..693fddf 100644
--- a/arch/arm/plat-omap/dss/dispc.c
+++ b/arch/arm/plat-omap/dss/dispc.c
@@ -832,7 +832,10 @@ void dispc_set_digit_size(int width, int height)
 int dispc_get_alpha_blending(enum omap_channel ch)
 {
 	u32 config;
+
+	enable_clocks(1);
 	config = dispc_read_reg(DISPC_CONFIG);
+	enable_clocks(0);
 
 	if (ch == OMAP_DSS_CHANNEL_LCD) {
 		if (config & 0x40000)
@@ -855,7 +858,10 @@ void dispc_get_color_keying(enum omap_channel ch, struct omap_color_key *key)
 	const struct dispc_reg tr_reg[] = {
 		DISPC_TRANS_COLOR0, DISPC_TRANS_COLOR1 };
 
+	enable_clocks(1);
 	config = dispc_read_reg(DISPC_CONFIG);
+	enable_clocks(0);
+
 	key->type = 0;
 
 	if (ch == OMAP_DSS_CHANNEL_LCD) {
@@ -871,8 +877,9 @@ void dispc_get_color_keying(enum omap_channel ch, struct omap_color_key *key)
 			key->enable = 1;
 		} else
 			key->enable = 0;
-
+		enable_clocks(1);
 		key->color = dispc_read_reg(tr_reg[ch]);
+		enable_clocks(0);
 	} else if (ch == OMAP_DSS_CHANNEL_DIGIT) {
 		/* check color keying is enabled or not */
 		if (config & 0x1000) {
@@ -887,7 +894,9 @@ void dispc_get_color_keying(enum omap_channel ch, struct omap_color_key *key)
 		} else
 			key->enable = 0;
 
+		enable_clocks(1);
 		key->color = dispc_read_reg(tr_reg[ch]);
+		enable_clocks(0);
 	}
 
 }
-- 
1.6.0.3

